samples/ios/Continote/Continote/EditNoteViewController.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  Copyright (c) 2017 Google Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import UIKit
+import Firebase
+import MaterialComponents.MaterialSnackbar
+
+/**
+ This ViewController presents the user with a Note editor to edit a specific Note, if
+ said Note is found for the user.
+ */
+class EditNoteViewController: BaseViewController {
+
+  // The key from the Firebase Realtime Database of the Note the user wants to edit.
+  // The Note the user wants to edit will be gathered from the Firebase Realtime Database once the
+  // user is confirmed to be signed in below. This is gathered separately on this screen,
+  // rather than passed from MyNotesViewController, to ensure the user is presented with the
+  // latest value for the Note (at the time of opening this screen).
+  var keyOfNoteToEdit: String!
+
+  // Firebase Realtime Database reference for the current user's Note to edit, based on the key.
+  private var noteToEditFirebaseDatabaseRef: DatabaseReference?
+
+  // UI outlets
+  @IBOutlet var temporaryTodoLabel: UILabel!
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+
+    // Set up this screen's AppBar title.
+    title = Constants.Text.screenTitle
+
+    // Style all labels.
+    temporaryTodoLabel.applyAppTheme(for: .normalText)
+  }
+
+  override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
+
+    guard keyOfNoteToEdit != nil else {
+      // The key for the Note to edit must be set for the user to be on this screen.
+      // This should never happen, but just in case.
+      navigationController?.popViewController(animated: true)
```

Comment:
Yup it does work! The edge case should never happen, but when it does, the EditNoteViewController slides off screen to show the MyNotesViewController.

I'm sure we could come up with a better user experience, but I think it's sufficient for this sample and it ensures the databaseKey is set at the time of the ViewController being on screen. And, of course, it should never happen in the first place.